core/src/test/java/org/elasticsearch/search/suggest/CompletionSuggestionOptionTests.java

```diff
@@ -0,0 +1,93 @@
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.suggest;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.text.Text;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.search.InternalSearchHitTests;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.suggest.completion.CompletionSuggestion;
+import org.elasticsearch.search.suggest.completion.CompletionSuggestion.Entry.Option;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.elasticsearch.common.xcontent.XContentHelper.toXContent;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertToXContentEquivalent;
+
+public class CompletionSuggestionOptionTests extends ESTestCase {
+
+    public static Option createTestItem() {
+        Text text = new Text(randomAsciiOfLengthBetween(5, 15));
+        int docId = randomInt();
+        int numberOfContexts = randomIntBetween(0, 3);
+        Map<String, Set<CharSequence>> contexts = new HashMap<>();
+        for (int i = 0; i < numberOfContexts; i++) {
+            int numberOfValues = randomIntBetween(0, 3);
+            Set<CharSequence> values = new HashSet<>();
+            for (int v = 0; v < numberOfValues; v++) {
+                values.add(randomAsciiOfLengthBetween(5, 15));
+            }
+            contexts.put(randomAsciiOfLengthBetween(5, 15), values);
+        }
+        SearchHit hit = null;
+        float score = randomFloat();
+        if (randomBoolean()) {
+            hit = InternalSearchHitTests.createTestItem(false);
+            score = hit.getScore();
+        }
+        Option option = new CompletionSuggestion.Entry.Option(docId, text, score, contexts);
+        option.setHit(hit);
+        return option;
+    }
+
+    public void testFromXContent() throws IOException {
+        Option option = createTestItem();
+        XContentType xContentType = randomFrom(XContentType.values());
+        boolean humanReadable = randomBoolean();
+        BytesReference originalBytes = toXContent(option, xContentType, humanReadable);
+        Option parsed;
```

Comment:
Something I did in ElasticsearchExceptionTests is to randomly shuffle the fields, just to be sure that parsers do not rely on the order of fields:

```
if (randomBoolean()) {
 try (XContentParser parser = createParser(xContentType.xContent(), originalBytes)) {
   originalBytes = shuffleXContent(parser, randomBoolean()).bytes();
 }
}
```

would you be OK to add something like this?